src/ucai/core/databricks.py

```diff
@@ -84,6 +89,15 @@ def _try_get_spark_session_in_dbr() -> Any:
         return
 
 
+def _is_in_databricks_notebook_environment() -> bool:
```

Comment:
This is a bit different than https://github.com/mlflow/mlflow/blob/7b16d762d21b44791f097ffaae101a9dc7600a53/mlflow/utils/databricks_utils.py#L175 
But as long as this field is enough for checking `in notebook` then we are good!

Reply:
Yeah the logic within MLflow is a bit intense (it has to be to support usage in so many different runtime environments). For serverless notebooks and classic compute, this simple check should be sufficient for development work of users. 
We may end up revisiting this if users start trying to do jobs-based notebook execution of dynamic UC function creation, but until someone complains about that, I think we should leave this as lightweight as possible :)